src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/ValidatorUtils.java

```diff
@@ -25,21 +25,34 @@ public final class ValidatorUtils {
 
     private ValidatorUtils() {}
 
-    /** maxLatencyValidator method check if the maxLatency and
-     * maxLatencyTier2 are valid.
+    /**
+     * maxLatencyValidator method checks if the maxLatency and maxLatencyTier2 are valid.
      * @param maxLatency maxLatency property
      * @param maxLatencyTier2 maxLatencyTier2 property
      * @throws InvalidFlowException invalidFlow exception
      */
-    public static void maxLatencyValidator(Long maxLatency, Long maxLatencyTier2) throws InvalidFlowException {
-        if (maxLatencyTier2 == null) {
+    public static void validateMaxLatencyAndLatencyTier(Long maxLatency, Long maxLatencyTier2) throws InvalidFlowException {
+        if (maxLatency == null && maxLatencyTier2 == null) {
             return;
         }
         if (maxLatency == null) {
             throw new InvalidFlowException(
                     "maxLatencyTier2 property cannot be used without maxLatency",
                     ErrorType.DATA_INVALID);
         }
+        if (maxLatencyTier2 == null || maxLatencyTier2.equals(0L)) {
+            return;
+        }
+
+        if (maxLatency < 0) {
+            throw new InvalidFlowException("maxLatency cannot be negative", ErrorType.DATA_INVALID);
+        }
+        if (maxLatencyTier2 < 0) {
+            throw new InvalidFlowException("maxLatencyTier2 cannot be negative", ErrorType.DATA_INVALID);
+        }
+        if (maxLatency.equals(0L) && maxLatencyTier2.equals(0L)) {
```

Comment:
if maxLatencyTier2 == 0 we will exit from method in line 44. So this `if` is always false

Reply:
Yeah, thanks! Somehow, I committed a wrong version, checkstyle is failing as well 🤷‍♂️. That block on lines 43-35 actually allows maxLatency to be negative, when tier2 is not set. I remember making it correct, but here we go again 😒